src/interchange/src/protobuf.rs

```diff
@@ -209,9 +214,9 @@ impl Decoder {
             })?,
             &mut packer,
         )?;
-        if let Some(pos) = position {
-            packer.push(Datum::from(pos))
-        };
+        if push_metadata {
+            packer.push(Datum::from(position));
```

Comment:
is it expected that position could be `None` here/would that represent some programming error on our part?

Reply:
the positions have historically always been optional, but I think by now maybe they're always filled in indeed. Not sure which connectors wouldn't have them.

Anyway, I will think of a better way to handle line numbers and positions as part of a separate diff, but I don't really want to trace through all that as part of this one.